google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/facet/AppEngineStandardFacet.java

```diff
@@ -52,4 +56,51 @@ public static AppEngineStandardFacet getAppEngineFacetByModule(@Nullable Module
     return FacetManager.getInstance(module).getFacetByType(ID);
   }
 
+  /**
+   * Returns {@code true} if this app is targeting the Flexible Environment instead of the Standard
+   * Environment.
+   */
+  public boolean isFlexEnvironment() {
+    AppEngineStandardWebApp appEngineStandardWebApp = getAppEngineStandardWebXml();
+    return appEngineStandardWebApp != null
+        && "flex".equalsIgnoreCase(appEngineStandardWebApp.getEnv().getStringValue());
+  }
+
+  /**
+   * Returns {@code true} if this app is targeting the Managed VM environment instead of the
+   * Standard Environment.
+   */
+  public boolean isManagedVmEnvironment() {
+    AppEngineStandardWebApp appEngineStandardWebApp = getAppEngineStandardWebXml();
+    return appEngineStandardWebApp != null
+        && "true".equalsIgnoreCase(appEngineStandardWebApp.getVm().getStringValue());
+  }
+
+  /**
+   * Returns {@code true} if this app is targeting some environment other than the Standard
+   * Environment.
+   */
+  public boolean isNonStandardEnvironment() {
+    return isFlexEnvironment() || isManagedVmEnvironment();
+  }
+
+  public boolean isJava8Runtime() {
+    AppEngineStandardWebApp appEngineStandardWebApp = getAppEngineStandardWebXml();
+    return appEngineStandardWebApp != null
+        && "java8".equalsIgnoreCase(appEngineStandardWebApp.getRuntime().getStringValue());
+  }
+
+  @Nullable
+  public AppEngineStandardWebApp getAppEngineStandardWebXml() {
```

Comment:
Do you think it would make sense to modify the asset provider to return AppEngineStandardWebApp instead of XmlFile given these changes?